cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -715,6 +727,7 @@ func (a *StaticAutoscaler) filterOutYoungPods(allUnschedulablePods []*apiv1.Pod,
 // ExitCleanUp performs all necessary clean-ups when the autoscaler's exiting.
 func (a *StaticAutoscaler) ExitCleanUp() {
 	a.processors.CleanUp()
+	a.DebuggingSnapshotter.Cleanup()
```

Comment:
`Cleanup()` resets the underlying snapshot, which isn't really necessary during CA shutdown. At the same time, it is not doing anything about a HTTP request that might be hanging.